Add setUndeocrated/setResizable APIs to Graphics.

bgroenks96 opened the pull request:
Add implemenations for all backends.

Note: most desktop backend implementations require a call to 'setWindowedMode' in order for the changes to take effect.

It is worth noting that this is not necessary with the JogAmp backend. However, as of the time of writing, that backend is not included in LibGDX master.

Fixes #3773

Tom-Ski commented:
Thanks for the contribution, unfortunately this can no longer be automatically merged.  If you can fix up  the merge conflict and use the LibGDX formatter that would be great.  There are a lot of unnecessary changes in these files which makes it a little difficult to review.

Its probably also worth adding to the javadoc which official backends are actually supported by this feature.

bgroenks96 commented:
@Tom-Ski Fixed with bfd5b80. Apologies for the resyncs.

I ran the auto-formatter on all of the changed files, so there may still be some spurious formatting changes if any of the source in those files was inconsistent with the formatting spec beforehand.

Tom-Ski commented:
Yikes that got out of hand.  Avoid running the formatting on the whole class that you changed a small line of code in.  In general only run the formatter on the code that you have changed or added.

bgroenks96 commented:
@Tom-Ski Would you like me to revert it?

bgroenks96 commented:
@Tom-Ski I got rid of the formatting commit. Some of the spurious whitespace stripping is still there in the original commit... but I'm hoping you can read past that? Does everything else look ok?

Tom-Ski commented:
Yeah its kind of a pain when it does that with whitespace, but it looks good, just the javadoc has some copy and paste error for the setResizable method. If you could fix that up and squash the commits we can merge.

bgroenks96 commented:
Done! Removed the erroneous note about Android from `setResizable` javadoc and squashed the changes in  30a7d2b. Let me know if there's anything else!

Tom-Ski commented:
Thanks!
